wolvic/browser/vr/wvr_manager.cc

```diff
@@ -491,15 +495,21 @@ bool WvrManager::SubmitFrameInternal(int16_t frame_index) {
     externalRect.width = 0.5f;
     externalRect.height = 1.0f;
   }
+
+  last_frame_index_ = frame_index;
   PushState(true);
   PullState([this]() {
-    return (system_state_.displayState.lastSubmittedFrameId >=
+    return (system_state_.displayState.lastSubmittedFrameId ==
             last_frame_index_) ||
            system_state_.displayState.suppressFrames ||
            !system_state_.displayState.isConnected;
   });
 
-  last_frame_index_ = system_state_.displayState.lastSubmittedFrameId;
+  // Avoid racing texture between processing in chromium and consuming in
+  // wolvic.
+  layer.textureHandle = 0;
+  PushState(true);
```

Comment:
I think I don't get this part. How do we avoid a race by setting the textureHandle to 0. Isn't that a valid value for a handle?

Also what about `last_frame_index_` ? Why don't we have to update it?

Reply:
Wolvic treats texture handle 0 as not valid and doesn't interfere with the textures.
If we keep the last texture handle value, the texture could be accessed during Tick(=vsync) on the Wolvic side. So, Resetting the texture handle value after frame processing can avoid accessing it.

`last_frame_index_` is moved up to line 499 to change the condition `==` instead of `>=`.